shared/src/main/scala/NumberHelpers.scala

```diff
@@ -51,9 +51,11 @@ object NumberHelpers:
   end gamma
 
   def multiplicity(a: VNum, b: VNum): VNum =
+    if a == VNum(0) || b == VNum(0) then return VNum(0)
+    if b.vabs == VNum(1) then return a.vabs
     var result = 0
     var current = a.toInt
-    while current % b.toInt == 0 do
+    while VNum(current) % VNum(b.toInt) == VNum(0) do
```

Comment:
Since we're working with 32-bit ints here, is it necessary to use `VNum()`?

Reply:
Yes. It really is. They're cast to VNum for the `%` behaviour.

Default scala modulo doesn't support floating point numbers and is truncated. We need floored floating point modulo

Reply:
Ah, right. In that case, we might as well make `current` and the rest `VNum`s too, so we're not restricted to 32-bit ints. `b.toIntegral` should work for that (although we should only call it once and store it in a variable)

Reply:
Maybe something like
```scala
  def multiplicity(a: VNum, b: VNum): VNum =
    if a == VNum(0) || b == VNum(0) then return VNum(0)
    if b.vabs == VNum(1) then return a.vabs
    var result = 0
    var current = a.toIntegral
    val bInt = b.toIntegral
    while current % bInt == VNum(0) do
      result += 1
      current /= bInt
    result
```

Reply:
Actually, I think nothing should be cast to int. I was reading the current v2 implementation and it doesn't have any cast to int

Reply:
Just leave them all as `VNum`. No need to uncast. I probably only keep the to integral because I was keeping some of the original code from before

Reply:
So maybe something like this?
```scala
  def multiplicity(a: VNum, b: VNum): VNum =
    if a == VNum(0) || b == VNum(0) then return VNum(0)
    if b.vabs == VNum(1) then return a.vabs
    var result = 0
    var current = a
    while current % b == VNum(0) do
      result += 1
      current /= b
    result
```

Reply:
Exactly like that